src/Modules/UngradedModules.jl

```diff
@@ -743,46 +795,104 @@ end
+end
 
-SubModuleOfFreeModule(F::FreeMod{R}, singular_module::Singular.smodule) where {R} = SubModuleOfFreeModule{R}(F, singular_module)
+@doc Markdown.doc"""
+    SubModuleOfFreeModule(F::FreeMod{R}, gens::Vector{<:FreeModElem}, default_ordering::ModuleOrdering) where {R}
+
+Construct the submodule of `F` generated by the elements of `gens` (the elements of 
+`gens` must live in `F`, if not, an execption is thrown). 
+Moreover, set the default ordering to `default_ordering`.
+"""
+function SubModuleOfFreeModule(F::FreeMod{R}, gens::Vector{<:FreeModElem}, default_ordering::ModuleOrdering) where {R}
+  @assert all(x -> parent(x) === F, gens)
+  return SubModuleOfFreeModule(F, ModuleGens(gens, F, default_ordering))
+end
+
+function SubModuleOfFreeModule(F::FreeMod{R}, singular_module::Singular.smodule) where {R} 
+  return SubModuleOfFreeModule(F, ModuleGens(F, singular_module))
+end
 
 @doc Markdown.doc"""
     SubModuleOfFreeModule(F::FreeMod{R}, gens::ModuleGens{R}) where {R}
 
 Construct the submodule of `F` generated by `gens`.
 """
-SubModuleOfFreeModule(F::FreeMod{R}, gens::ModuleGens{R}) where {R} = SubModuleOfFreeModule{R}(F, gens)
+function SubModuleOfFreeModule(F::FreeMod{R}, gens::ModuleGens{R}) where {R} 
+  subModule = SubModuleOfFreeModule{R}(F)
+  subModule.gens = gens
+  return subModule
+end
 
 @doc Markdown.doc"""
     SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L}
 
 Construct the submodule generated by the rows of `A`. The embedding free
 module is `F`. In particular, `rank(F) == ncols(A)` must hold.
 """
-SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L} = SubModuleOfFreeModule{L}(F, A)
+function SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L} 
+  subModule = SubModuleOfFreeModule{L}(F)
+  O = [FreeModElem(sparse_row(A[i,:]), F) for i in 1:nrows(A)]
+  subModule.gens = ModuleGens(O, F)
+  subModule.matrix = A
+  return subModule
+end
+
+@doc Markdown.doc"""
+    SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}, default_ordering::ModuleOrdering) where {L}
+
+Construct the submodule generated by the rows of `A`. The embedding free
+module is `F`. In particular, `rank(F) == ncols(A)` must hold.
+Morevoer, set the default ordering to `default_ordering`.
+"""
+function SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}, default_ordering::ModuleOrdering) where {L} 
+  subModule = SubModuleOfFreeModule{L}(F)
+  O = [FreeModElem(sparse_row(A[i,:]), F) for i in 1:nrows(A)]
+  subModule.gens = ModuleGens(O, F, default_ordering)
+  subModule.matrix = A
+  return subModule
+end
 
 @doc Markdown.doc"""
     SubModuleOfFreeModule(A::MatElem{L}) where {L} 
 
 Construct the submodule generated by the rows of `A`.
 
+!!! note
+  The ambient free module of the submodule is constructed by the function and therefore
+  not compatible with free modules that are defined by the user or by other functions. 
+  For compatibility, use `SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L}`.
```

Comment:
I think admonitions must be indented with 4 spaces, and there must be an empty line in between (see https://github.com/JuliaLang/julia/issues/45622)
```suggestion
!!! note

    The ambient free module of the submodule is constructed by the function and therefore
    not compatible with free modules that are defined by the user or by other functions. 
    For compatibility, use `SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L}`.
```